test/functional/test_framework/wallet.py

```diff
@@ -55,25 +80,41 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
+    def get_utxos(self, amount):
```

Comment:
Exposing `self._utxos` via a getter weakens the existing guarantees of the class (i.e. the list of utxos is not writeable from the outside). It didn't seem like an appropriate change to make. (At the very least, someone made a deliberate choice not to expose this field and I didn't want to undo that.)

Reply:
I see if that's the case its fine. Just out of curiosity when you said, "Someone made a delibrate choice...", is there a specific PR or dev notes that you can link where getter method has been discarded

Reply:
Just for the sake of abstraction, so that outside code doesn't access the inner "organs" of MiniWallet, I didn't write a getter for the private member `_utxos`.

As pointed out by @danben the "list of utxos is not writeable from the outside"

Reply:
aah i see that makes sense. Thank you!